src/presence/HeartbeatComponent.js

```diff
@@ -0,0 +1,63 @@
+// @flow strict-local
+import { PureComponent } from 'react';
+import { AppState } from 'react-native';
+import type { Auth, Dispatch } from '../types';
+
+import { connect } from '../react-redux';
+import { tryGetAuth } from '../account/accountsSelectors';
+import { reportPresence } from '../actions';
+import Heartbeat from './heartbeat';
+
+type Props = $ReadOnly<{|
+  dispatch: Dispatch,
+  auth: Auth | void,
+|}>;
+
+/**
+ * Component providing a recurrent `presence` signal.
+ */
+// Note that "PureComponent" is of questionable veracity: this component's
+// entire purpose is to emit network calls for their observable side effects.
+// However, it is at least true that there is never a need to call `render()` if
+// the props haven't changed.
+class PresenceHeartbeat extends PureComponent<Props> {
+  /** Callback for Heartbeat object. */
+  onHeartbeat = (state: boolean) => {
+    // N.B.: If `auth` changes, we do not send out a final `false` presence
+    // status for the previous `auth`. It's the responsibility of our logout
+    // handler to determine whether that's necessary.
+    //
+    // (TODO: ensure that our logout handlers actually do that.)
+    if (this.props.auth) {
+      this.props.dispatch(reportPresence(state));
+    }
+  };
+
+  heartbeat: Heartbeat = new Heartbeat(this.onHeartbeat, 1000 * 60);
+
+  componentDidMount() {
+    this.onStateChange();
+    AppState.addEventListener('change', this.onStateChange);
+  }
+
+  componentWillUnmount() {
+    AppState.removeEventListener('change', this.onStateChange);
+    this.heartbeat.stop();
```

Comment:
This looks asymmetrical. Should we be starting it in `componentDidMount`?

... Ah I see, we do that conditionally via `onStateChange`.

How about naming the latter something like `updateHeartbeatState`? I.e. named for what it does rather than when it's called. There are a couple of call sites that aren't the "state change" callback, and this name is a little puzzling at both of them.

Reply:
Works for me. (The asymmetry is worth a couple of comments, too.)